PlusRemote/Logic/vtkSlicerPlusRemoteLogic.cxx

```diff
@@ -0,0 +1,580 @@
+    return;
+  }
+  // Update command attributes
+  this->CmdStopVolumeReconstruction->SetCommandAttribute("VolumeReconstructorDeviceId", volumeReconstructorDeviceID);
+  this->CmdStopVolumeReconstruction->SetCommandAttribute("OutputVolFilename", outputFilename);
+  this->CmdStopVolumeReconstruction->SetCommandAttribute("OutputVolDeviceName", outputDevice);
+  // Send command
+  this->ExecuteCommand(this->CmdStopVolumeReconstruction, connectorNode, callback);
+}
+
+//---------------------------------------------------------------------------
+void vtkSlicerPlusRemoteLogic::SaveConfigFile(vtkMRMLIGTLConnectorNode* connectorNode, std::string configFileName, vtkCallbackCommand* callback)
+{
+  if (!connectorNode)
+  {
+    return;
+  }
+  // Update command attributes
+  this->CmdSaveConfig->SetCommandAttribute("Filename", configFileName);
+  // Send command
+  this->ExecuteCommand(this->CmdSaveConfig, connectorNode, callback);
+}
+
+//---------------------------------------------------------------------------
+void vtkSlicerPlusRemoteLogic::UpdateTransform(vtkMRMLIGTLConnectorNode* connectorNode, vtkMRMLLinearTransformNode* transformNode, vtkCallbackCommand* callback)
+{
+  if (!connectorNode || !transformNode)
+  {
+    return;
+  }
+
+  // Get transform matrix as string
+  vtkMatrix4x4* transformMatrix = transformNode->GetMatrixTransformToParent();
+
+  std::stringstream transformSS;
+
+  for (int i = 0; i < 4; ++i)
+  {
+    for (int j = 0; j < 4; ++j)
+    {
+      transformSS << transformMatrix->GetElement(i, j);
+      if (i != 3 || j != 3)
+      {
+        transformSS << " ";
+      }
+    }
+  }
+
+  // Get transform date as string
+  std::string transformDate = this->GetTimestampString("%Y-%m-%d %H:%M:%S.f");
+  std::string transformValue = transformSS.str();
+
+  // Update command attributes
+  this->CmdUpdateTransform->SetCommandAttribute("TransformName", transformNode->GetName());
+  this->CmdUpdateTransform->SetCommandAttribute("TransformValue", transformValue);
+  this->CmdUpdateTransform->SetCommandAttribute("TransformDate", transformDate);
+  // Send command
+  this->ExecuteCommand(this->CmdUpdateTransform, connectorNode, callback);
+}
+
+//---------------------------------------------------------------------------
+void vtkSlicerPlusRemoteLogic::ExecuteCommand(vtkSlicerOpenIGTLinkCommand* command, vtkMRMLIGTLConnectorNode* connectorNode, vtkCallbackCommand* callback)
+{
+  command->SetCommandAttribute("Name", command->GetCommandName());
+  command->RemoveAllObservers();
+  if (callback)
+  {
+    command->AddObserver(vtkSlicerOpenIGTLinkCommand::CommandCompletedEvent, callback);
+  }
+  connectorNode->SendCommand(command);
+}
+
+//---------------------------------------------------------------------------
+std::string vtkSlicerPlusRemoteLogic::GetTimestampString(std::string format)
+{
+  // Get the string representing the current time.
+  // Format specified by input string.
+  // Default format is: YearMonthDay_HourMinuteSecond.
```

Comment:
common convention is to write default argument values in the cxx files as comments:

std::string vtkSlicerPlusRemoteLogic::GetTimestampString(std::string format /*  = "%Y%m%d_%H%M%S" */ )

I would recommend to follow this convention and remove this comment line.